osu.Game/Beatmaps/IBeatmapMetadataInfo.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using osu.Framework.Localisation;
+
+#nullable enable
+
+namespace osu.Game.Beatmaps
+{
+    /// <summary>
+    /// Metadata representing a beatmap. May be shared between multiple beatmap difficulties.
+    /// </summary>
+    public interface IBeatmapMetadataInfo : IEquatable<IBeatmapMetadataInfo>
+    {
+        /// <summary>
+        /// The romanised title of this beatmap.
+        /// </summary>
+        string Title { get; }
+
+        /// <summary>
+        /// The unicode title of this beatmap.
+        /// </summary>
+        string TitleUnicode { get; }
+
+        /// <summary>
+        /// The romanised artist of this beatmap.
+        /// </summary>
+        string Artist { get; }
+
+        /// <summary>
+        /// The unicode artist of this beatmap.
+        /// </summary>
+        string ArtistUnicode { get; }
+
+        /// <summary>
+        /// The author of this beatmap.
+        /// </summary>
+        string Author { get; } // eventually should be linked to a persisted User.
+
+        /// <summary>
+        /// The source of this beatmap.
+        /// </summary>
+        string Source { get; }
+
+        /// <summary>
+        /// The tags of this beatmap.
+        /// </summary>
+        string Tags { get; }
+
+        /// <summary>
+        /// The time in milliseconds to begin playing the track for preview purposes.
+        /// If -1, the track should begin playing at 40% of its length.
+        /// </summary>
+        int PreviewTime { get; }
+
+        /// <summary>
+        /// The filename of the audio file consumed by this beatmap.
+        /// </summary>
+        string AudioFile { get; }
+
+        /// <summary>
+        /// The filename of the background image file consumed by this beatmap.
+        /// </summary>
+        string BackgroundFile { get; }
+
+        /// <summary>
+        /// A user-presentable display title representing this metadata.
+        /// </summary>
+        string DisplayTitle
```

Comment:
Not sure about all these default implementations. On the one hand it makes sense from the POV of being able to override these members' implementations, but on the other hand they seem like abuse and maybe extension methods would fit better.

Reply:
Will change to extension methods, I think we want to avoid interface defaults everywhere at this point.